api/tests/test_asset_log.py

```diff
@@ -229,6 +229,78 @@ def test_authenticated_admin_user_get_filtered_list_of_asset_logs(
             f"{self.test_other_asset.serial_number} - {self.test_other_asset.asset_code}",
         )
 
+    @patch("api.authentication.auth.verify_id_token")
+    def test_authenticated_admin_user_get_asset_logs_filtered_by_serial_number(
+        self, mock_verify_id_token
+    ):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = (
+            f"{self.asset_logs_url}/?asset_serial={self.test_other_asset.serial_number}"
+        )
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data["results"]), 1)
```

Comment:
checking the response length can be flaky - if a new asset log is created in the base class setup this would break. could we check that all items in the result contain the serial number the filter was for?

Reply:
If a new asset-log is created in the base class setup, many tests would break. Tests are suppose to be stand alone. Test objects created in the base class are to be shared by all or most test methods. It is for this reason I've created an asset-log in the test method just for this assertion and thus checking for the response length.

Reply:
I think checking that all items in the result contain the serial number won't scale because the returned asset object in the log is just a string representation and wasn't serialized.

Reply:
@egichuri